transform_aes.c

```diff
@@ -162,28 +167,23 @@ static ssize_t aes_choose_rx_sa( transop_aes_t * priv, const u_int8_t * peer_mac
  */
 static size_t aes_best_keysize(size_t numBytes)
 {
-    if (numBytes >= AES256_KEY_BYTES )
-    {
+	/* this is a shortcut to always use full key size for security reasons */
+	/* however, performance needs might require platform-dependant changes, e.g. for routers */
         return AES256_KEY_BYTES;
-    }
-    else if (numBytes >= AES192_KEY_BYTES)
-    {
-        return AES192_KEY_BYTES;
-    }
-    else
-    {
-        return AES128_KEY_BYTES;
-    }
 }
 
 /** The aes packet format consists of:
  *
  *  - a 8-bit aes encoding version in clear text
  *  - a 32-bit SA number in clear text
- *  - ciphertext encrypted from a 32-bit nonce followed by the payload.
+ *  - a 128 bit inner message authentication code (last encrypted data block encrypted again using K2),
+ *  - the separately encrypted IV (using K2),
+ *  - the 16-bit packet length of the following data (encrypted together with...
+ *  - ...the following payload data (using K1).
+ *  with K1 = key1 and K2 = key2.
  *
- *  [V|SSSS|nnnnDDDDDDDDDDDDDDDDDDDDD]
- *         |<------ encrypted ------>|
+ *  [V|SSSS|MMMMMMMMMMMMMMMM|IIIIIIIIIIIIIIII|LLDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD]
+ *         |< encrypted K2 >|< encrypted K2 >|<------- encrypted K1 --------->|
```

Comment:
Replace "encrypted K2" with "encrypted with K2" or similar to avoid confusion

Reply:
I will think of something.